#! /bin/sh /usr/share/dpatch/dpatch-run
## longer-test-timeout.dpatch by  <weasel@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: extend timeout for mutex test.  cf. Tor#6227

@DPATCH@
diff -urNad '--exclude=CVS' '--exclude=.svn' '--exclude=.git' '--exclude=.arch' '--exclude=.hg' '--exclude=_darcs' '--exclude=.bzr' tor~/src/test/test_util.c tor/src/test/test_util.c
--- tor~/src/test/test_util.c	2012-06-24 15:16:32.000000000 +0200
+++ tor/src/test/test_util.c	2012-06-24 16:09:56.360489077 +0200
@@ -1213,7 +1213,7 @@
     if (strmap_get(_thread_test_strmap, "thread 1") &&
         strmap_get(_thread_test_strmap, "thread 2")) {
       done = 1;
-    } else if (time(NULL) > started + 25) {
+    } else if (time(NULL) > started + 250) {
       timedout = done = 1;
     }
     tor_mutex_release(_thread_test_mutex);
